File: android/app/src/main/java/com/polypulse/app/domain/model/Market.kt
package com.polypulse.app.domain.model

import kotlinx.serialization.Serializable

@Serializable
data class Market(
    val id: String,
    val question: String,
    val imageUrl: String?,
    val volume: Double,
    val outcomes: List<Outcome>,
    val tags: List<String> = emptyList()
)

@Serializable
data class Outcome(
    val name: String,
    val price: Double, // 0.0 to 1.0
    val tokenId: String
)

File: backend/app/models/market.py
from pydantic import BaseModel
from typing import List, Optional

class Token(BaseModel):
    token_id: str
    outcome: str
    price: float
    winner: bool = False

class Market(BaseModel):
    condition_id: str
    question: str
    tokens: List[Token]
    volume: float = 0.0
    tags: List[str] = []
    market_slug: str = ""
